src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -116,13 +110,34 @@ public synchronized void disposeAsync() {
         new Thread("Disposing "+base) {
             @Override
             public void run() {
-                for (File dir : tbr)
+                for (File dir : tbr) {
+                    LOGGER.info(() -> "deleting " + dir);
                     try {
-                        new FilePath(dir).deleteRecursive();
-                    } catch (IOException | InterruptedException e) {
-                        e.printStackTrace();
+                        delete(dir.toPath());
+                    } catch (IOException e) {
+                        LOGGER.log(Level.WARNING, null, e);
                     }
+                }
             }
         }.start();
     }
+
+    private void delete(Path p) throws IOException {
+        LOGGER.fine(() -> "deleting " + p);
+        if (Files.isDirectory(p, LinkOption.NOFOLLOW_LINKS)) {
+            try (DirectoryStream<Path> children = Files.newDirectoryStream(p)) {
+                for (Path child : children) {
+                    delete(child);
+                }
+            }
+        }
+        try {
+            Files.deleteIfExists(p);
+        } catch (DirectoryNotEmptyException x) {
+            try (DirectoryStream<Path> children = Files.newDirectoryStream(p)) {
+                throw new IOException(Iterables.toString(children), x);
+            }
```

Comment:
NIT:  could we replace guava by calling Files.list(p) (which is a `java.util.Stream` and collecting the results?